### DISCUSSION TITLE
Rotate with 'Background' interpolation and 'Point' filter produces empty result

### DISCUSSION THREAD

**Discussion by jk-keller** [ImageMagick/ImageMagick#7419] [Help]
### ImageMagick version

7.1.1-33 Q16-HDRI aarch64 22263

### Operating system, version and so on

macOS Sonoma 14.5, MacBook Pro M1 Max

### Description

When rotating an image using the `Background` interpolation with the `Point` filter, the result is an empty image. Every other interpolation & filter combo seems to work with rotation. 

![01--mario](https://github.com/ImageMagick/ImageMagick/assets/25354950/925795b4-1163-4529-89ef-41f8bbead8ea)

`magick 01--mario.png -interpolate Background -filter Point -rotate 10 01--mario-bg_pt.png`

![01--mario-bg_pt-rotate](https://github.com/ImageMagick/ImageMagick/assets/25354950/bae3a675-a708-4765-a96b-6e42220acc90)

Resizing the image using `Background` & `Point` does work though:
`magick 01--mario.png -interpolate Background -filter Point -resize 1024x200! 01--mario-bg_pt-resize.png`

![01--mario-bg_pt-resize](https://github.com/ImageMagick/ImageMagick/assets/25354950/dfe27680-7844-4f39-b0bc-163bc12d6106)

Anyone know why this may happen?

**Comment by urban-warrior** [comment]
Background interpolation replaces the source pixel with the background color.   You are getting expected results.  @fmw42, can you verify?

**Reply by fmw42** [reply]
**@urban-warrior**  I do not understand -interpolate background.  I get the same results as the OP. If I specify -background green1 before -interpolate background, the rotated image is all green, but the resized image shows no change other than the resize. I see no green there.  What is the point of replacing the whole image with background color using -rotate?

```
All white as expected as the default background color is white
magick 01--mario.png -interpolate Background -filter Point -rotate 10 01--mario-bg_pt.png

All green1
magick 01--mario.png -background green1 -interpolate Background -filter Point -rotate 10 01--mario-bg_pt.png

No color change to white
magick 01--mario.png -interpolate Background -filter Point -resize 1024x200! 01--mario-bg_pt-resize.png

No color change to green1 
magick 01--mario.png -background green1 -interpolate Background -filter Point -resize 1024x200! 01--mario-bg_pt-resize.png
```

**Reply by urban-warrior** [reply]
See https://imagemagick.org/Usage/misc/#interpolate_simple:

One other simple interpolation method is also provided, 'Background' which simply returns the current background color for any 'sampling' of the source image. In many ways this is rather usless, as typically you will just end up with a blank solid colored image. Its primary use is as a check of more complex resampling functions, such as a Resampling Failure, where the EWA resampling filter (typically used from the General Distortion Operator) will fall back to an interpolated lookup when resampling fails to find any pixels in its 'support' or resampling area. By setting interpolation to 'Background', and setting a background color to something that stands out (like 'red') you can then look for pixels in the resulting image to see where resampling 'failed' or 'missed all source image pixels' for some reason or another. Typically due to too small a support setting, or from playing with expert filter options.

**Reply by fmw42** [reply]
Thanks, but I don't see why it does not produce a Green image for me when using -resize.

**Reply by urban-warrior** [reply]
Resizing an image is not responsive to the `-interpolate` option.  Rotating is responsive.

**Reply by fmw42** [reply]
Thanks.  Perhaps that should be mentioned somewhere -- perhaps on Anthony's page at the link you specified above.  Any other interpolation options that are not responsive for either -rotate or -resize (and I presume -scale and -sample, as well).

**Reply by urban-warrior** [reply]
Post a 'pull' request from this repo: https://github.com/ImageMagick/Usage, with your suggested changes and we'll approve the edits.

**Reply by fmw42** [reply]
OK, but you did not answer my question.

Are there any other interpolation options that are not responsive for either -rotate or -resize (and I presume -scale and -sample, as well).

**Reply by urban-warrior** [reply]
We don't track which operators utilize interpolation and which do not.  You can inspect the source code @ `MagickCore/*.{c,h}` and look for these methods: `InterpolatePixelChannel()`, `InterpolatePixelChannels()`, and `InterpolatePixelInfo()`.

**Reply by jk-keller** [reply]
Sorry to be wading into territory I only have vague understanding of, but part of my original question remains: if expected behavior of `-Background` is to replace source with background color, why is `-filter Point` the outlier when comes to outputs? Using @fmw42's example with all the other filters produces the following:

```
magick 01--mario.png -background green1 -interpolate Background -filter Bartlett -rotate 10 01--mario-gr_bg_bar.png
magick 01--mario.png -background green1 -interpolate Background -filter Blackman -rotate 10 01--mario-gr_bg_bla.png
# repeat with all other filters...
magick 01--mario.png -background green1 -interpolate Background -filter XXXXXXXXX -rotate 10 01--mario-gr_bg_xxx.png
```

![all_filters](https://github.com/ImageMagick/ImageMagick/assets/25354950/ab5b1d2d-f612-4e04-9d47-e097e354496f)

**Reply by urban-warrior** [reply]
The background interpolate operator is considered an experimental feature. It is primarily used for debugging specific operations, offering insights into distortion or resampling algorithms. However, it is not generally useful for all purposes. Point filtering results in a single pixel, therefore the final result mirrors the background color. In contrast, other filters employ blending, where the background color contributes only a small percentage to the final color.
